GettingStarted.md

```diff
@@ -0,0 +1,214 @@
+    - [Fallback URL](#fallback-url)
+- [Examples](#examples)
+- [Other Resources](#other-resources)
+  - [Articles](#articles)
+  - [Videos](#videos)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Why Should I Use a Service Worker?
+
+You have a web app, and you'd like it to load quickly and work offline.
+
+You'd like to use proven tools to handle the details for you, to work around
+common gotchas and follow best practices.
+
+# Terminology
+
+## Service Worker
+
+A [service worker](http://www.html5rocks.com/en/tutorials/service-worker/introduction/)
+is a background script that intercepts network requests made by your web app.
+It can use the [Cache Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Cache)
+to respond to those requests.
+
+You do not have to write your own service worker script; this guide will explain
+how to generate one customized for your web app.
+
+## App Shell
+
+"App Shell" refers to the local resources that your web app needs to load its
+basic structure. This will always include some HTML, and will likely also
+include CSS and JavaScript, either inline or in external files.
+
+Some static web apps consist entirely of an App Shell.
+
+A helpful analogy when thinking about your App Shell is to the code and
+resources that would be published to an App Store for a native iOS or Android
+application.
+
+The App Shell should ideally be loaded directly from the local cache, just like
+a native iOS or Android application is loaded directly from a device's storage.
+
+## Dynamic Content
+
+The dynamic content is all of the data, images, and other resources that your
+web app needs to function, but exists independently from your App Shell.
+
+Sometimes this data will come from external, third-party APIs, and sometimes
+this will be first-party data that is dynamically generated or frequently
+updated.
+
+For example, if your web app is for a newspaper, it might make use of a
+first-party API to fetch recent articles, and a third-party API to fetch the
+current weather. Both of those types of requests fall into the category of
+"dynamic content".
+
+Unlike the App Shell, dynamic content is usually ephemeral, and it's important
+to choose the right caching strategy for each source of dynamic content.
+
+## Caching Strategy
+
+There are several approaches to caching, outlined in excellent detail in the
+[The Offline Cookbook](https://jakearchibald.com/2014/offline-cookbook/).
+
+You should always use a [cache-first strategy](https://jakearchibald.com/2014/offline-cookbook/#cache-falling-back-to-network)
+for your App Shell. `sw-precache` handles the details of that for you.
+
+The right caching strategy for your dynamic content is not always clear-cut.
+It's recommended that you read through the
+[The Offline Cookbook](https://jakearchibald.com/2014/offline-cookbook/) and
+think about how important freshness is for each of your data sources.
+
```

Comment:
Please take another crack at this section. You both repeat and contradict yourself.